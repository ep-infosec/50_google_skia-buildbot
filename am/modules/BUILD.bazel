load("//infra-sk:index.bzl", "karma_test", "ts_library")

karma_test(
    name = "am_test",
    src = "am_test.ts",
    deps = [
        ":am_ts_lib",
        "@npm//@types/chai",
        "@npm//@types/sinon",
        "@npm//chai",
        "@npm//sinon",
    ],
)

ts_library(
    name = "am_ts_lib",
    srcs = ["am.ts"],
    visibility = ["//visibility:public"],
    deps = [
        "//am/modules/json:index_ts_lib",
        "@npm//common-sk",
        "@npm//lit-html",
    ],
)
