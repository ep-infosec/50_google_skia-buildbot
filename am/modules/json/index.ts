// DO NOT EDIT. This file is automatically generated.

export interface Note {
	text: string;
	author: string;
	ts: number;
}

export interface Incident {
	key: string;
	id: string;
	active: boolean;
	start: number;
	last_seen: number;
	params: Params;
	notes: Note[] | null;
}

export interface Silence {
	key: string;
	active: boolean;
	user: string;
	param_set: ParamSet;
	created: number;
	updated: number;
	duration: string;
	notes: Note[] | null;
}

export interface RecentIncidentsResponse {
	incidents: Incident[] | null;
	flaky: boolean;
	recently_expired_silence: boolean;
}

export interface StatsRequest {
	range: string;
}

export interface Stat {
	num: number;
	incident: Incident;
}

export interface IncidentsResponse {
	incidents: Incident[] | null;
	ids_to_recently_expired_silences: { [key: string]: boolean } | null;
}

export interface IncidentsInRangeRequest {
	range: string;
	incident: Incident;
}

export interface AuditLog {
	id: string;
	action: string;
	user: string;
	body: string;
	timestamp: number;
}

export type Params = { [key: string]: string };

export type ParamSet = { [key: string]: string[] };

export type StatsResponse = (Stat | null)[] | null;
