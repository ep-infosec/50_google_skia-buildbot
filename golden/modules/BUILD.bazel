load("//infra-sk:index.bzl", "karma_test", "sass_library", "ts_library")

karma_test(
    name = "common_test",
    src = "common_test.ts",
    deps = [
        ":common_ts_lib",
        "//golden/modules/search-controls-sk",
        "//infra-sk/modules:test_util_ts_lib",
        "@npm//@types/chai",
        "@npm//chai",
        "@npm//common-sk",
    ],
)

ts_library(
    name = "common_ts_lib",
    srcs = ["common.ts"],
    visibility = ["//visibility:public"],
    deps = [
        ":rpc_types_ts_lib",
        "//golden/modules/search-controls-sk",
        "@npm//common-sk",
    ],
)

ts_library(
    name = "demo_util_ts_lib",
    srcs = ["demo_util.ts"],
    visibility = ["//visibility:public"],
)

ts_library(
    name = "rpc_types_ts_lib",
    srcs = ["rpc_types.ts"],
    visibility = ["//visibility:public"],
)

ts_library(
    name = "settings_ts_lib",
    srcs = ["settings.ts"],
    visibility = ["//visibility:public"],
)

ts_library(
    name = "shared_demo_data_ts_lib",
    srcs = ["shared_demo_data.ts"],
    visibility = ["//visibility:public"],
    deps = ["@npm//common-sk"],
)

sass_library(
    name = "colors_sass_lib",
    srcs = ["colors.scss"],
    visibility = ["//visibility:public"],
)
