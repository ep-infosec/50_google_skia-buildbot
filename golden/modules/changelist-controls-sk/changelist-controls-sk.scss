@import 'node_modules/elements-sk/colors.scss';

changelist-controls-sk {
  font-family: sans-serif;

  .info {
    display: flex;
    justify-content: space-between;
    max-width: 60em;

    .title {
      font-size: 1em;
      font-weight: bold;
      text-transform: capitalize;
    }
  }

  .inputs {
    display: flex;
    margin-top: 4px;
    margin-bottom: 8px;

    & > * {
      /* centers the elements vertically*/
      margin: auto 0;
    }

    .spacer {
      flex-grow: 1;
      max-width: 4em;
    }

    .radiogroup {
      display: flex;
      border: 1px solid var(--light-gray);
      border-radius: 5px;
      margin-left: 4px;
    }
  }

  .tryjob-container {
    display: flex;
    flex-wrap: wrap;

    .tryjob {
      border: 1px solid var(--gray);
      padding: 0.5em;
      border-radius: 5px;
      margin: 0.25em;

      a {
        color: var(--black);
      }
    }
  }


}
