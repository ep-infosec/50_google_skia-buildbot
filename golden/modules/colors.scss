/* Common Gold Colors via CSS Variables.

 Can be imported via:

    @import '../colors';
*/

html {
  --gold: rgb(217, 95, 2);
  --bright-orange: rgb(255, 127, 0);

  --light-error-red: rgb(255, 230, 230);
  --transparent-white: rgba(255, 255, 255, 0);

  --triaged-positive: #1B9E77;
  --triaged-negative: #E7298A;
  --untriaged: #A6761D;
}
