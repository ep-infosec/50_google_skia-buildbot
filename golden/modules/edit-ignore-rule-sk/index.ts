import './edit-ignore-rule-sk';
