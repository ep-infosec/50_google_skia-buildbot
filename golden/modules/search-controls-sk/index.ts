import './search-controls-sk';

export { SearchCriteriaToHintableObject, SearchCriteriaFromHintableObject } from './search-controls-sk';
