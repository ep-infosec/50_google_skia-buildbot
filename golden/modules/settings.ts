/**
 * This file lists helpers for returning global constants in Gold. These should not change once
 * an instance has booted up, so they need not be returned via JSON requests.
 *
 * Settings are expected to be in window.GoldSettings and the functions here are nice helpers
 * for that, so as to demystify "where do these values come from?"
 */

export interface GoldSettings {
  title?: string;
  defaultCorpus?: string;
  baseRepoURL?: string;
}

function getSettings(): GoldSettings | undefined {
  return (window as any).GoldSettings as GoldSettings | undefined;
}

export function title(): string {
  return getSettings()?.title || '';
}

export function defaultCorpus(): string {
  return getSettings()?.defaultCorpus || '';
}

export function baseRepoURL(): string {
  return getSettings()?.baseRepoURL || '';
}

export function testOnlySetSettings(newSettings: GoldSettings) {
  (window as any).GoldSettings = newSettings;
}
