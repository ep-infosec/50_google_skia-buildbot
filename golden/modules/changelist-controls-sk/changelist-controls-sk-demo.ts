import './index';

import { $$ } from 'common-sk/modules/dom';
import { ChangelistControlsSk } from './changelist-controls-sk';
import { twoPatchsets } from './test_data';

const ele = $$<ChangelistControlsSk>('changelist-controls-sk');
ele!.summary = twoPatchsets;
