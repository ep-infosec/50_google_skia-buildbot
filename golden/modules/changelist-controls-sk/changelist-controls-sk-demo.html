<!DOCTYPE html>
<html>
  <head>
    <title>changelist-controls-sk</title>
    <meta charset="utf-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <style>
      .search_response {
        border: 1px solid black;
        padding: 1em 1em 0 1em;
        margin: 1em 1em 1em 0;
        width: 1000px;
      }
      .search_stats {
        margin-bottom: 1em;
      }
    </style>
  </head>
  <body class="body-sk">
    <!-- This boilerplate is from the search page that embeds this.-->
    <div class="search_response">
      <div class="search_stats">
        Displaying: 10 of 58 images starting at 20<br />
        UniqueTests Visible: 8 <br />
        UniqueTests Total: 19 <br />
      </div>
      <changelist-controls-sk></changelist-controls-sk>
    </div>
  </body>
</html>
